sympy/core/expr.py

```diff
@@ -2962,10 +2962,8 @@ def series(self, x=None, x0=0, n=6, dir="+", logx=None, cdir=0):
 
         if x0 in [S.Infinity, S.NegativeInfinity]:
             sgn = 1 if x0 is S.Infinity else -1
-            s = self.subs(x, sgn/x).series(x, n=n, dir='+', cdir=cdir)
-            if n is None:
-                return (si.subs(x, sgn/x) for si in s)
-            return s.subs(x, sgn/x)
+            s = self.subs(x, sgn*x).aseries(x, n).subs(x, sgn*x)
```

Comment:
Many functions have proper power series expansions at infinity; rational functions, for example, have expansions everywhere. Those should usually be created by `nseries`. Asymptotic series may be located anywhere in the extended complex plane, not necessarily at infinity. `aseries` should be called only when a power series does not exist.